src/store-api/src/region_engine.rs

```diff
@@ -583,6 +583,60 @@ impl RegionStatistic {
     }
 }
 
+/// The response of syncing the manifest.
+#[derive(Debug)]
+pub enum SyncManifestResponse {
+    NotSupported,
+    Mito {
+        synced: bool,
+    },
+    Metric {
+        metadata_synced: bool,
+        data_synced: bool,
+        new_opened_logical_region_ids: Vec<RegionId>,
+    },
+}
+
+impl SyncManifestResponse {
+    /// Returns true if new manifest is installed.
+    pub fn synced(&self) -> bool {
+        match self {
+            SyncManifestResponse::NotSupported => false,
+            SyncManifestResponse::Mito { synced } => *synced,
+            SyncManifestResponse::Metric {
+                metadata_synced,
+                data_synced,
+                ..
+            } => *metadata_synced || *data_synced,
```

Comment:
A bit confusing, why it is `or` but not `and`

Reply:
We need some documents for the behavior. Maybe we should also comment when the region is considered sync above the enum variant definitions.

Reply:
added in [cf4fc5a](https://github.com/GreptimeTeam/greptimedb/pull/5878/commits/cf4fc5a5cf4118988545c04892de6777b42272a6)